frontends/web/src/routes/device/bitbox02/passphrase.tsx

```diff
@@ -35,345 +36,360 @@ const INFO_STEPS_DISABLE = 0;
-          alertUser(this.props.t('genericError'));
+          alertUser(t('genericError'));
           return;
         }
         const { mnemonicPassphraseEnabled } = result.deviceInfo;
-        this.setState({
-          // before enabling/disabling we show 1 or more pages to inform about the feature
-          // each page has a continue button that jumps to the next or finally toggles passphrase
-          // infoStep counts down in decreasing order
-          infoStep: mnemonicPassphraseEnabled
-            ? INFO_STEPS_DISABLE
-            : INFO_STEPS_ENABLE,
-          passphraseEnabled: mnemonicPassphraseEnabled,
-        });
-      })
-      .catch(console.error);
-  }
+        setPassphraseEnabled(mnemonicPassphraseEnabled);
+        setInfoStep(mnemonicPassphraseEnabled ? INFO_STEPS_DISABLE : INFO_STEPS_ENABLE);
+      }
+    }).catch(console.error);
 
-  private togglePassphrase = async () => {
-    const { deviceID, t } = this.props;
-    const enable = !this.state.passphraseEnabled;
-    this.setState({ status: 'progress' });
+    return () => {
+      // In case the effect runns multiple times, we only conisder the last result,
+      // and ignore the prior ones.
+      ignoreResult = true;
+    };
+  }, [deviceID, setPassphraseEnabled, setInfoStep, t]);
+
+  // Toggle Passphrase on device.
+  const togglePassphrase = async () => {
+    setStatus('progress');
     try {
-      const result = await setMnemonicPassphraseEnabled(deviceID, enable);
+      // !undefined is true...maybe we shoudl check that passphraseEnabled
+      // is not undefined? It should not be possible here though.
+      const result = await setMnemonicPassphraseEnabled(deviceID, !passphraseEnabled);
       if (!result.success) {
-        route(`/settings/device-settings/${deviceID}`);
+        navigate(`/settings/device-settings/${deviceID}`);
         alertUser(t(`passphrase.error.e${result.code}`, {
           defaultValue: result.message || t('genericError'),
         }));
         return;
       }
       const deviceInfoResult = await getDeviceInfo(deviceID);
       if (deviceInfoResult.success) {
-        this.setState({
-          passphraseEnabled: deviceInfoResult.deviceInfo.mnemonicPassphraseEnabled,
-          status: 'success',
-        });
+        setPassphraseEnabled(deviceInfoResult.deviceInfo.mnemonicPassphraseEnabled);
+        setStatus('success');
       }
     } catch (error) {
       console.error(error);
     }
   };
-
-  private stopInfo = () => route(`/settings/device-settings/${this.props.deviceID}`);
-
-  private continueInfo = () => {
-    if (this.state.infoStep === 0) {
-      this.togglePassphrase();
-      return;
-    }
-    this.setState(({ infoStep }) => ({ infoStep: infoStep - 1 }));
-  };
-
-  private backInfo = () => {
-    if (this.state.infoStep === undefined) {
+  // Navigate back to settings.
+  const stopInfo = () => navigate(`/settings/device-settings/${deviceID}`);
+  // Back button click event.
```

Comment:
is this comment really useful here and some previous funcitons? 
To me it is an indication that the variable name `backInfo` could be more descriptive, so that the comment is not needed.
 maybe just `handleBack`  instead of `backInfo`, or `handleAbort` instead of `stopInfo`.
keep as is for now, lets change it later in refactoring commit.

Reply:
got it, yes in the refactoring commit it is still `stopInfo` I did rename it to `onClose` but I felt like that was weird. But I like `handleBack`